xbrowse/datastore/elasticsearch_datastore.py

```diff
@@ -258,42 +264,99 @@ def get_elasticsearch_variants(
+                    genotype_filters[encoded_sample_id] = [('term', genotype_filter)]
+                elif '$gte' in genotype_filter:
+                    genotype_filter = {k.replace("$", ""): v for k, v in genotype_filter.items()}
+                    genotype_filters[encoded_sample_id] = [('range', genotype_filter)]
+                elif "$in" in genotype_filter:
+                    num_alt_values = genotype_filter['$in']
+                    genotype_filters[encoded_sample_id] = [('term', num_alt_value) for num_alt_value in num_alt_values]
 
-            s = s.filter(atleast_one_nonref_genotype_filter)
+        encoded_sample_ids = [_encode_name(family_individual_ids_to_sample_ids.get(indiv_id) or indiv_id)
+                              for indiv_id in (indivs_to_consider or [])]
 
+        min_ab = None
+        min_gq = None
         if quality_filter is not None and indivs_to_consider:
-            #'vcf_filter': u'pass', u'min_ab': 17, u'min_gq': 46
             min_ab = quality_filter.get('min_ab')
             if min_ab is not None:
-                min_ab /= 100.0   # convert to fraction
+                min_ab /= 100.0  # convert to fraction
             min_gq = quality_filter.get('min_gq')
             vcf_filter = quality_filter.get('vcf_filter')
-            for indiv_id in indivs_to_consider:
-                sample_id = family_individual_ids_to_sample_ids.get(indiv_id) or indiv_id
-                encoded_sample_id = _encode_name(sample_id)
+            if vcf_filter is not None:
+                s = s.filter(~Q('exists', field='filters'))
 
-                #'vcf_filter': u'pass', u'min_ab': 17, u'min_gq': 46
-                if min_ab:
-                    s = s.filter(
-                        ~Q('term', **{encoded_sample_id+"_num_alt": 1}) |
-                        Q('range', **{encoded_sample_id+"_ab": {'gte': min_ab}}))
-                    #logger.info("### ADDED FILTER: " + str({encoded_sample_id+"_ab": {'gte': min_ab}}))
-                if min_gq:
-                    s = s.filter('range', **{encoded_sample_id+"_gq": {'gte': min_gq}})
-                    #logger.info("### ADDED FILTER: " + str({encoded_sample_id+"_gq": {'gte': min_gq}}))
-                if vcf_filter is not None:
-                    s = s.filter(~Q('exists', field='filters'))
-                    #logger.info("### ADDED FILTER: " + str(~Q('exists', field='filters')))
+        if is_nested:
+            #  Subquery for child docs with the requested sample IDs
+            sample_id_q = Q('terms', sample_id=encoded_sample_ids)
+
+            # Return inner hits with the requested samples
+            if indivs_to_consider:
+                s = s.filter(Q('has_child', type='genotype', query=sample_id_q, inner_hits={'size': len(indivs_to_consider)}))
+            else:
+                s = s.filter(Q('has_child', type='genotype', query=Q(), inner_hits={'size': 100}))
```

Comment:
why is the limit 100 and not 3 or 10? either way would create a global constant or maybe there's a way to get rid of calls where indivs_to_consider is None?

Reply:
Thats the elasticsearch max number of inner hits that can be returned, but you are right it should be a constant. And this is used by gene search so it needs to stay. As a general rule, I'm not making any improvements to this search since I'm writing a clean new search, but in that new search I will clean up gene search so it does explicitly include the sample ids it wants